File: C++/longest_palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> hashmap;
        map<char, int>::iterator iter;
        int size = 0;
        int add = 0;
        
        for(auto w : s) {
            if(hashmap.find(w) == hashmap.end()) hashmap[w] = count(s.begin(), s.end(), w);
            else continue;
        }
                
        for(iter = hashmap.begin(); iter != hashmap.end(); iter++) {
            if(iter->second % 2 == 0) size += iter->second;
            if(iter->second % 2 == 1) { 
                size += (iter->second - 1);
                add = 1;
            }
        }
        
        return size + add;
    }
};

File: Python3/longest_palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        hashmap = {}
        size = 0
        add = 0
        
        for w in s:
            if w not in hashmap: hashmap[w] = s.count(w)
            else: continue
                
        for i, k in enumerate(hashmap):
            if hashmap[k] % 2 == 0: size += hashmap[k]
            if hashmap[k] % 2 == 1: 
                size += (hashmap[k] - 1)
                add = 1
                
        return size + add
